doc/api/addons.md

```diff
@@ -1,6 +1,6 @@
 # C/C++ Addons
```

Comment:
Good point @mhdawson, worth changing this as well.

I get these matches with a quick (rip)grep, @mhdawson do the n-api ones need changing?

```bash
➜  node git:(master) ❯ rg "C/C\+\+ Addons"                                                                                                                                                   ~/wrk/com/node
doc/api/_toc.md
10:* [C/C++ Addons](addons.html)
11:* [C/C++ Addons - N-API](n-api.html)

doc/api/addons.md
1:# C/C++ Addons
234:section titled [C/C++ Addons - N-API](n-api.html).

doc/api/n-api.md
15:outlined in the section titled  [C/C++ Addons](addons.html).
```